File: ems-backend/src/main/java/com/employeemanagement/service/EmployeeService.java
package com.employeemanagement.service;

import java.util.List;

import com.employeemanagement.dto.EmployeeDTO;

public interface EmployeeService {
	//Create a new employee
	EmployeeDTO createEmployee(EmployeeDTO employeeDTO);
	
	//Retrieve an employee with given id
	EmployeeDTO getElementById(Long id);
	
	//Get all the employees present in the database.
	List<EmployeeDTO> getAllEmployee();
	
	//Update an existing employee.
	EmployeeDTO updateEmployee(Long id, EmployeeDTO updatedEmployee);
	
	//Delete Employee
	void deleteEmployee(Long id);
}

File: ems-frontend/src/Services/EmployeeService.js
import axios from "axios";

//Base url of our spring boot project
const REST_API_BASE_URL = "http://localhost:8080/employee";

//Get all the employees mapping
export const listEmployees = () => {
  //Format for gettign data from database to react json object using axios
  //axios.get(rest_api_url)
  return axios.get(REST_API_BASE_URL + "/getAllEmployee");
};

//Add new employee
export const createEmployee = (employee) => {
  //Format for sending json to java spring:
  //axios.post(rest_api_url, json_object);
  return axios.post(REST_API_BASE_URL + "/addEmployee", employee);
};

//Get employee by id
export const getEmployee = (id) => {
  //Format for get request by id
  //axios.get(url+ '/' + id)
  return axios.get(REST_API_BASE_URL + "/" + id);
};

//Update employee
export const updateEmployee = (id, employee) => {
  //Format for put request
  //axios.put(url_with_id, newEmplpyee object reference
  return axios.put(REST_API_BASE_URL + "/" + id, employee);
};

//Delete employee
export const deleteEmployee = (id) => {
  //Format for delete request
  return axios.delete(REST_API_BASE_URL + "/" + id);
};
